### PR TITLE
Modernize the package a bit

### PR DESCRIPTION
**Pull request by ocefpaf**
In this PR:
- update PyPI GHA
- moved as much as I can to pyproject.toml
- remove setuptools_scm_git_archive to rely only on the tags to publish new versions
- drop py37 from the CIs, this was already dropped by conda-forge
- we should not depend on esmpy until we can get a wheel on PyPI xref.: https://github.com/conda-forge/esmpy-feedstock/issues/72

### CODE REVIEW

**Comment by ocefpaf** [comment]
@huard I also want to implement linting with ruff but I'll leave that for another PR.

For now, we still cannot publish on PyPI b/c someone with admin rights to this repo needs to add the PyPI token to the repo secrets.

**Comment by huard** [comment]
Ok, give me 5 minutes, I'll get the token.

File: .github/workflows/pypi.yaml [hunk 1]
```diff
@@ -1,48 +1,55 @@
...
-
-      - name: upload artifacs
-        uses: actions/upload-artifact@v3
+      - name: Publish a Python distribution to PyPI
+        if: success() && github.event_name == 'release'
+        uses: pypa/gh-action-pypi-publish@release/v1
         with:
-          path: dist/*
+          user: __token__
+          password: ${{ secrets.PYPI_PASSWORD }}
```
Comment (huard) [inline .github/workflows/pypi.yaml:55]:
```suggestion
          password: ${{ secrets.PYPI_TOKEN }}
```

**Review by huard — APPROVED** [review]

**Comment by huard** [comment]
Thanks !

**Pull request merged**